src/stripe/tasks/get-plans-task.ts

```diff
@@ -0,0 +1,64 @@
+import { Stripe } from 'stripe';
+import { Actor, DatabaseTransactionHandler } from 'graasp';
+import { FastifyLoggerInstance } from 'fastify';
+import { Plan } from '../interfaces/plan';
+import { DEFAULT_PRICE, INDIVIDUAL_PLAN } from '../../util/constants';
+import { BaseStripeTask } from './base-stripe-task';
+
+export type GetPlansTaskInputType = {
+  product?: string;
+};
+
+export class GetPlansTask extends BaseStripeTask<Plan[]> {
+  get name(): string {
+    return GetPlansTask.name;
+  }
+
+  input: GetPlansTaskInputType;
+
+  constructor(member: Actor, stripe: Stripe, input?: GetPlansTaskInputType) {
+    super(member, stripe);
+    this.input = input ?? {};
+  }
+
+  async run(handler: DatabaseTransactionHandler, log: FastifyLoggerInstance): Promise<void> {
+    this.status = 'RUNNING';
+
+    // This block of code takes all the Individual plans (Organisations plans or similar may exist in the future)
+    // And then sort them by level, the level is a metadata used to rank the plans (ex. Free < Standard < Professional)
+    // All the additional informations linked to a plan are stored in the metadata
+    const plans = Array.from((await this.stripeService.prices.list({
+        product: this.input?.product ?? undefined,
+        expand: ['data.product']
+      })).data
+      .filter((price) => (<Stripe.Product>price.product).metadata['type'] === INDIVIDUAL_PLAN)
+      // This lines creates a Map containing all the productsId with their prices
+      // ex: { key: "prod_xxx", entry: [{ price1 }, { price2 }] }
+
+      // This would be a better solution when implmented in node.js: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/group
+      // this allows the following : (await prices).filter((price) => ).group((price) => (e.product as Stripe.Product)).map()
+      .reduce((entryMap, e) => entryMap.set((<Stripe.Product>e.product).id, [...entryMap.get((<Stripe.Product>e.product).id)||[], e]), new Map()))
+      .map(([key, prices]) => {
```

Comment:
Aren't we in nodejs ? why not use the commented solution?

Reply:
Because node.js doesn't implement it yet, as shown by the compatibility table in the link